addons/popochiu/engine/interfaces/i_room.gd

```diff
@@ -110,13 +117,197 @@ func clear_instances() -> void:
+		await get_tree().process_frame
+		E.in_room = true
+		
+		# Calling this will make the camera be set to its default values and will store the state of
+		# the main room (the last parameter will prevent Popochiu from changing the scene to the
+		# same that is already loaded)
+		goto_room(room.script_name, false, true, true)
+	
+	# Make the camera be ready for the room
+	current.setup_camera()
+	
+	# Update the core state
+	if E.loaded_game:
+		C.player = C.get_character(E.loaded_game.player.id)
+	else:
+		current.state.visited = true
+		current.state.visited_times += 1
+		current.state.visited_first_time = current.state.visited_times == 1
+	
+	# Add the PopochiuCharacter instances to the room
+	if (rooms_states[room.script_name]["characters"] as Dictionary).is_empty():
+		# Store the initial state of the characters in the room
+		current.state.save_characters()
+	
+	current.clean_characters()
+	
+	# Load the state of characters in the room
+	for chr_script_name: String in rooms_states[room.script_name]["characters"]:
+		var chr_dic: Dictionary = rooms_states[room.script_name]["characters"][chr_script_name]
+		var chr: PopochiuCharacter = C.get_character(chr_script_name)
+		
+		if not chr: continue
+		
+		chr.position = Vector2(chr_dic.x, chr_dic.y)
+		chr._looking_dir = chr_dic.facing
+		chr.visible = chr_dic.visible
+		chr.modulate = Color.from_string(chr_dic.modulate, Color.WHITE)
+		chr.self_modulate = Color.from_string(chr_dic.self_modulate, Color.WHITE)
+		chr.light_mask = chr_dic.light_mask
+		
+		current.add_character(chr)
+	
+	# If the room must have the player character but it is not part of its $Characters node, then
+	# add the PopochiuCharacter to the room
+	if (
+		current.has_player
+		and is_instance_valid(C.player)
+		and not current.has_character(C.player.script_name)
+	):
+		current.add_character(C.player)
+		# Place the PC in the middle of the room
+		C.player.position = Vector2(E.width, E.height) / 2.0
+		await C.player.idle()
+	
+	# Load the state of Props, Hotspots, Regions and WalkableAreas
+	for type in PopochiuResources.ROOM_CHILDS:
+		for script_name in rooms_states[room.script_name][type]:
+			var node: Node2D = current.callv(
+				"get_" + type.trim_suffix("s"),
+				[(script_name as String).to_pascal_case()]
+			)
+			var node_dic: Dictionary =\
+			rooms_states[room.script_name][type][script_name]
+			
+			for property in node_dic:
+				if not PopochiuResources.has_property(node, property): continue
+				
+				node[property] = node_dic[property]
+	
+	for c in get_tree().get_nodes_in_group("PopochiuClickable"):
+		c.room = current
+	
+	await current._on_room_entered()
+	
+	if E.loaded_game:
+		C.player.global_position = Vector2(
+			E.loaded_game.player.position.x,
+			E.loaded_game.player.position.y
+		)
+	
+	if _use_transition_on_room_change:
+		E.tl.play_transition(E.tl.FADE_OUT)
+		await E.tl.transition_finished
+		
+		await E.wait(0.3)
+	else:
+		await get_tree().process_frame
+	
+	if not current.hide_gi:
```

Comment:
Should this be called `hide_gui` ? I may be wrong

Reply:
Yes. Done.